### PR TITLE
Add new xml payload flow model node

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Part of [Implementing data flow nodes](https://github.com/wso2-enterprise/eggplant-project/issues/116).

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 1]
```diff
@@ -396,17 +398,32 @@ public void visit(VariableDeclarationNode variableDeclarationNode) {
...
+            }
+            if (initializerNode.kind() == SyntaxKind.XML_TEMPLATE_EXPRESSION) {
+                startNode(FlowNode.Kind.NEW_XML_PAYLOAD_DATA)
+                        .metadata()
+                        .description(NewXMLPayloadData.DESCRIPTION)
+                        .stepOut()
+                        .properties().expression(initializerNode);
+            } else {
+                startNode(FlowNode.Kind.NEW_DATA)
+                        .metadata()
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:413]: We can replace all these lines with a single visit method:

```java
@Override
public void visit(TemplateExpressionNode templateExpressionNode) {
    if (templateExpressionNode.kind() == SyntaxKind.XML_TEMPLATE_EXPRESSION) {
        startNode(FlowNode.Kind.NEW_XML_PAYLOAD_DATA)
                .metadata()
                .description(NewXMLPayloadData.DESCRIPTION)
                .stepOut()
                .properties().expression(templateExpressionNode);
    }
}
```

We also need to reevaluate whether we can use `new xml payload`. With the new proposed design, we are prioritizing the expression, and let the user decide how the variable assignment should be.

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java [hunk 2]
```diff
@@ -244,4 +246,17 @@ public static String getVariableName(Node node) {
...
+     * @param bracedExpressionNode the braced expression node
+     * @return the expression node
+     */
+    public static ExpressionNode getExpressionNodeFromBracedExpression(BracedExpressionNode bracedExpressionNode) {
+        ExpressionNode exception = bracedExpressionNode.expression();
+        if (exception.kind() == SyntaxKind.BRACED_EXPRESSION) {
+            return getExpressionNodeFromBracedExpression((BracedExpressionNode) exception);
+        }
+        return exception;
+    }
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java:261]: Isn't this redundant when using the visitor design pattern?

File: flow-model-generator/modules/flow-model-generator-ls-extension/src/test/resources/node_template/config/new_xml_payload.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        "optional": false,
+        "editable": true
+      },
+      "expression": {
+        "metadata": {
+          "label": "Expression",
+          "description": "Create new XML payload"
+        },
+        "valueType": "EXPRESSION",
+        "value": "xml``",
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-ls-extension/src/test/resources/node_template/config/new_xml_payload.json:42]: IMO, we shouldn't display the XML template expression syntax to the user as it is Ballerina-specific and not intended for an integration developer. Therefore, only the XML within the backtick string should be sufficient.

```suggestion
        "value": "",
```

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java (hunk 1 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:413]: As I understand the requirement new xml payload used to represent variable declarations with xml template literal values in L.H.S. If we move the visitor to the template expression node level, when we have xml template literal any place we will represent it as a new xml payload. Is that our requirement? @hasithaa

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java (hunk 2 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java:261]: updated

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java (hunk 1 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java]: updated

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 4]
```diff
@@ -367,11 +370,11 @@ private void checkForNewConnection(NewExpressionNode newExpressionNode,
         startNode(FlowNode.Kind.NEW_CONNECTION)
                 .metadata().description(nodeTemplate.metadata().description()).stepOut()
                 .codedata()
-                    .org(nodeTemplate.codedata().org())
-                    .module(nodeTemplate.codedata().module())
-                    .object(nodeTemplate.codedata().object())
-                    .symbol(nodeTemplate.codedata().symbol())
-                    .stepOut()
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:374]: These are intended formatting done by the IntelliJ for faceted builders. Let's preserve them

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 5]
```diff
@@ -384,6 +387,22 @@ private void checkForNewConnection(NewExpressionNode newExpressionNode,
         }
     }
 
+    @Override
+    public void visit(BracedExpressionNode bracedExpressionNode) {
+        bracedExpressionNode.expression().accept(this);
+    }
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:393]: I think this is implicitly done by the visitor pattern.

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java (hunk 2 above)
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java:261]: Do we need this method?

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java (hunk 5 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java]: removed

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java (hunk 2 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CommonUtils.java]: removed

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java (hunk 4 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java]: updated

**Review by nipunayf — APPROVED** [review]

**Pull request merged**